synapse/_scripts/register_new_matrix_user.py

```diff
@@ -20,11 +20,20 @@
 import hmac
 import logging
 import sys
-from typing import Callable, Optional
+from typing import Any, Callable, Optional
 
 import requests
 import yaml
 
+_CONFLICTING_SHARED_SECRET_OPTS_ERROR = """\
+Conflicting options 'registration_shared_secret' and 'registration_shared_secret_path'
+are both defined in config file.
+"""
```

Comment:
Kind of a shame that we need to do the same checks that we have in `RegistrationConfig`. I don't think there's an easy way around that though.